docs/plugins.rst

```diff
@@ -301,6 +301,43 @@ Here is a basic example of how to set up your own writer::
         signals.get_writer.connect(add_writer)
 
 
+Modifying Generators to Inject Content
+---------------------------------------
+
+You can programatically inject articles or pages using plugins. This can be 
+useful if you plan to fetch articles from an API, for example. 
+
+Here is a very simple example of how we can build a plugin that injects
+custom articles, using the article `article_generator_pretaxonomy` signal::
+
+    from pelican import signals
+    from pelican.contents import Article
+    from pelican.readers import BaseReader
+    import datetime
+
+    def addArticle(articleGenerator):
+        settings = articleGenerator.settings
+
+        # Author, category, and tags are objects, not strings, so they need to
+        # be handled using BaseReader's process_metadata.
+        baseReader = BaseReader(settings) 
+
+        content = "I am the body of an injected article!"
+
+        newArticle = Article(content, {
+            "title": "Injected Article!",
+            "date": datetime.datetime.now(),
+            "category": baseReader.process_metadata('category', 'fromAPI'),
+            "tags": baseReader.process_metadata("tags", 'tagA, tagB')
```

Comment:
Random switch between single-quoted and double-quoted strings is triggering me a bit :).